pkg/kexec/kexec_load_linux.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2015-2019 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package kexec
+
+import (
+	"fmt"
+	"syscall"
+	"unsafe"
+
+	"golang.org/x/sys/unix"
+)
+
+// Load loads the given segments into memory to be executed on a kexec-reboot.
+//
+// It is assumed that segments is made up of the next kernel's code and text
+// segments, and that `entry` is the entry point, either kernel entry point or trampoline.
+//
+// Load will align segments to page boundaries and deduplicate overlapping ranges.
+func Load(entry uintptr, segments []Segment, flags uint64) error {
+	for i := range segments {
+		segments[i] = AlignPhys(segments[i])
+	}
+
+	segments = Dedup(segments)
+	ok := false
+	for _, s := range segments {
```

Comment:
Put this loop in a "func Contains(s []Segment, p uintptr) bool" function.

Reply:
Done